src/main/scala/org/scalameter/Measurement.scala

```diff
@@ -1,15 +1,19 @@
 package org.scalameter
 
+import org.scalameter.picklers.Pickler
 import scala.collection.Seq
 
 @SerialVersionUID(-2541697615491239986L)
-case class Measurement(value: Double, params: Parameters, data: MeasurementData, units: String) {
-  def complete: Seq[Double] = data.complete
+case class Measurement[T: Pickler](value: T, params: Parameters,
+  data: MeasurementData[T], units: String) {
+  private[scalameter] def pickler: Pickler[T] = implicitly[Pickler[T]]
+
+  def complete: Seq[T] = data.complete
   def success: Boolean = data.success
-  def errors: Errors = new Errors(this)
-  def failed = this.copy(data = data.copy(success = false))
+  def errors(implicit num: Numeric[T]): Errors[T] = new Errors(this)
```

Comment:
I've embedded Numeric as an implicit in the `errors` method and not in the `Measurement` since that should be more robust when dealing with more complex types, like Map[String, Double]
However, this method isn't currently used, so maybe we could remove it?